airootfs/usr/share/vx-img/flash-image.sh

```diff
@@ -42,45 +69,55 @@ if [[ $_surface == 0 && $_haskeys == 0 ]]; then
     fi
 fi
 
-echo "Flashing a new image to the hard disk. This will destroy any existing data on the disk. Continue? [y/N]:"
+clear
 
-read -r answer
+data=$(lsblk -x SIZE -nblo NAME,LABEL,SIZE,TYPE | grep "Data" | awk '{ print $1 }')
 
-if [[ $answer != 'y' && $answer != 'Y' ]]; then
-    exit
-fi
+if [[ -n $data ]]; then
+    _datadisk="$data"
+else 
+    readarray disks < <(lsblk -x SIZE -nblo NAME,LABEL,SIZE,TYPE | grep "disk" | awk '{ print $1 }')
+    # This dumps the newlines at the end of the entries in the lsblk table
+    disks=("${disks[@]//$'\n'/}")
 
-clear
+    while true; do 
+        unset answer
+        menu "${disks[@]}" "Which disk contains the data to flash?"
+        if [[ $err == 1 ]]; then
+            echo "Something went wrong. Please try again."
+            exit
+        fi
 
-IFS=$'\n' read -r -d '' -a disks <<< "$(lsblk -x SIZE -nblo NAME,LABEL,SIZE,TYPE | grep "disk" | awk '{print $1}')"
+        if [[ -n $answer ]]; then
+            selected="${disks[answer-1]}"
 
-while true; do 
-    i=1
-    for disk in "${disks[@]}"; do
-        echo "$i. /dev/$disk"
-        ((i+=1))
+            if [[ -z $selected ]]; then
+                echo "Invalid selection, starting over"
+                #clear
+                continue
+            fi
+            _datadisk="/dev/$selected"
+        else
+            _datadisk="/dev/${disks[-1]}"
+        fi
+        break
     done
-    echo "Which disk contains the data to flash? [${disks[-1]}]" 
-    read -r answer
-    if [[ -n $answer ]]; then
-        selected=${disks[answer-1]}
 
-        if [[ -z $selected ]]; then
-            echo "Invalid selection, starting over"
-            clear
-            continue
-        fi
-        _datadisk="/dev/$selected"
-    else
-        _datadisk="/dev/${disks[-1]}"
-    fi
-    break
-done
+    mount "${_datadisk}3" /mnt
```

Comment:
so does this mean it mounts the third partition no matter what device you're choosing? Even if there is more then one and you select a different device?

Reply:
That's correct. This is a hold over from earlier when I was assuming the data disk was a Ventoy disk formatted the way we format them. I left it because I'm not completely sure how to handle arbitrary partition schemes. After the disk is selected, I guess we could present a menu for the partitions on it (skipped if it only has one partition).

Reply:
@carolinemodic I wrote in a partition selection step, so this should be a non-issue now.